### DISCUSSION TITLE
PrivateBin and Ubuntu Upgrade Issue

### DISCUSSION THREAD

**Discussion by chrish112** [PrivateBin/PrivateBin#1286] [Q&A]
### Describe the problem/question

Hi Community,
I started with PrivateBin 1.6.2 on Ubuntu Live Server 20.04.05 a bit ago. I wanted to upgrade both the server itself and then PrivateBin and what I noticed was that my website was working fine after the upgrade but when I tried to edit some of the folders to pull configuration files via WinSCP, obviously. I edited the permissions of the folder to the account that allows me to remote via WinSCP but even then I was still denied. I tried to edit only the specific folder I needed, for the bootstrap.php, and now the website is totally offline. I double checked the permissions and even went to the very top, the entire folder that houses PrivateBin and recursively changed permissions to the www-data account which I had before. Nothing, PrivateBin was not finding any of the files anymore and I have to think either Ubuntu upgrade to 22.04 broke my ability to EDIT permissions and for software to follow those directions.
I say this because PrivateBin 1.6.2 on upgraded Ubuntu 22.04 functioned, it was only when I started to edit permissions that the software completely broke. 

Has anyone seen this? Attempted to do upgrades?

### Did you use the FAQ section?

- [X] Yes, I have read [the FAQ](https://github.com/PrivateBin/PrivateBin/wiki/FAQ) and I found no solution/answer there.

### What you did?

1. Upgrade Ubuntu from 20.04.05 live server to Ubuntu 22.04 LTS
2. Edit permissions on folders in PrivateBin for configuration backups prior to upgrading Privatebin 1.6.2 to PrivateBin1.7.1
...

### What happens

The website goes down, provides me not a HTTP or HTTPS error but a message below:

```
<?php
/**
 * PrivateBin
 *
 * a zero-knowledge paste bin
 *
 * @link      https://github.com/PrivateBin/PrivateBin
 * @copyright 2012 SĆ©bastien SAUVAGE (sebsauvage.net)
 * @license   https://www.opensource.org/licenses/zlib-license.php The zlib/libpng License
 * @version   1.6.2
 */

// change this, if your php files and data is outside of your webservers document root
define('PATH', '');

define('PUBLIC_PATH', __DIR__);
require PATH . 'vendor' . DIRECTORY_SEPARATOR . 'autoload.php';
new PrivateBin\Controller;
```

### What should happen

Editing permissions on the folder using an admin account to allow me to use winscp to copy files. Then I set it back and the site should continue to run.

### Issue reproducibility

No, I cannot reproduce it on https://privatebin.net.

**Comment by elrido** [comment]
That clearly means that your webserver is no longer forwarding the request to your php-fpm or php-fcgi daemon or you no longer have mod-php loaded/installed. We can not support you with your individual webserver setup. I would recommend looking at the webserver and system logs, check the configuration changes, test with other php applications (i.e. a simple test.php containing just `<?php phpinfo();` ), etc. to figure out what got borked.

Alternatively, if you don't want to maintain your own webserver configuration, consider using [one of](https://github.com/PrivateBin/docker-unit-alpine) our [container images](https://github.com/PrivateBin/docker-nginx-fpm-alpine) that comes with a preconfigured webserver and will be easier to update (just change the image tag and re-start the container).

**Reply by chrish112** [reply]
Hi elrido,
Thank you for responding to my post. Still relatively new to getting into the deep end with Linux as I only picked it up in December. I knew of the PrivateBin application as I used it in the past but on the end-user's side once it was already setup. I have been using and implemented it where I work and have made multiple adjustments but specifically this error I was unable to correct. As for the php-fpm or php-fcgi daemon I do not recall specifically installing that but I know I installed php module alongside apache2. I ran through the initial setups and double checked my work and everything seems to be in order, obviously something is off. 

I followed your recommendation for the test.php and it works fine. There's something wrong with the webserver accessing the main privatebin folder and then the next steps. It defaulted to the index.php but I can go through my URL and specify the tpl folder or go into bootstrap.php. Although it is not displaying correctly the information is there. None of the documentroot folders are outside of the path. It seems that it just stopped understanding what to do once it went into the privatebin folder. As for specifying the path I don't think I need to specify the path if all of the contents are inside the privatebin folder.

At this point I'm going to load a new VM with Ubuntu 22.04 and try the latest 1.7.1 and see how the install goes.
